src/instance.rs

```diff
@@ -332,19 +325,15 @@ where
     where
         T: PyClass<Frozen = True> + Sync,
     {
-        let cell = self.get_cell();
-        // SAFETY: The class itself is frozen and `Sync` and we do not access anything but `cell.contents.value`.
-        unsafe { &*cell.get_ptr() }
+        // SAFETY: The class itself is frozen and `Sync`.
+        unsafe { &*self.get_class_object().get_ptr() }
     }
 
-    pub(crate) fn get_cell(&'py self) -> &'py PyCell<T> {
-        let cell = self.as_ptr().cast::<PyCell<T>>();
+    pub(crate) fn get_class_object(&self) -> &PyClassObject<T> {
+        let class_object = self.as_ptr().cast::<PyClassObject<T>>();
         // SAFETY: Bound<T> is known to contain an object which is laid out in memory as a
```

Comment:
```suggestion
        // SAFETY: Bound<T: PyClass> is known to contain an object which is laid out in memory as a
```
It might be worth including that this is only valid for `T: PyClass` (which is true here, because we have the bound on the imp), same for the other one